pkg/component/watch.go

```diff
@@ -16,6 +16,24 @@ import (
 	"github.com/pkg/errors"
 )
 
+func isIgnore(path string, patterns []string) bool {
```

Comment:
need tests

Reply:
please add comment explaining that this does and what arguments are

Reply:
Ack